Style, Refactor, Feature updates/fixes

yassirog opened the pull request:
## What does this change?

Resolves issue #62 , #73 , #76 , #77 , #78 

**#62 Expand Semester 4 Sprints**
I made the Semester 4 Sprints bigger, with JS logic. I made them bigger, because the sprints are not just 3 weeks like other sprints. I've made a variable and put sprint 19 and 20 in it. I used a ```.join('  ')``` to add a class to the sprints. Only the 19th and 20th sprint had a adjustment to their height, so another class than the others.

**#73 Aligned text with expand agenda button**
I've removed the flex-direction column in the class of the label, because of this the text was next to the button. Also added align-text center so the text was aligned in the center. 

#76 Refactored Code
I've refactored every file to fix the consistency in the code and unnecessary things, for example I deleted commented code and fixed ugly code like: ```display:flex;``` (correct is: ```display: flex;``` )

#77 Components for SVG
I deleted every SVG in a file and made components for them. I copied the SVG and the styling and dropped it in the SVG. This ensures that you have good overview in your code.

#78 2 Same components
There was 2 SVG's (An arrow) 1 pointing to the left and 1 pointing to the right. I deleted the right one and just kept the left one. I gave the arrow svg a prop so i could manipulate the css for it for the right one. I've added the class .right-arrow that has: ``` transform: scaleX(-1); ``` which makes the arrow looking to the right direction, because of this you could just use 1 SVG instead of 2.

## How Has This Been Tested?

- [ ] User test
- [x] Accessibility test
- [x] Performance test
- [x] Responsive Design test
- [ ] Device test
- [ ] Browser test

Accessibility & Performance test are done by google's lighthouse test.
Responsive test i've done my self the inspect in google and put the layout on responsive and slide it left and right to see if the site breaks and it the results were good.

## Images
Accessibility test & Performance Test
![image](https://github.com/user-attachments/assets/2a21e6ac-e43f-4a12-a5ba-1144e01bcc76)

### Before & After of Semester 4
![image](https://github.com/user-attachments/assets/69ba69b3-cc16-4027-98cc-a1e21b2e6441)
![image](https://github.com/user-attachments/assets/72f0604e-ce8c-427d-8e66-5e614d902180)


## How to review

1. Double check code
2. Merge if everything is good

<!-- Provide instructions to help others verify the change. This could take the form of "On PROD, do X and witness Y. On this branch, do X and witness Z. " -->


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **New Features**
  - Introduced a new arrow icon for navigation controls.
  - Added a refreshed, responsive logo that now enhances the header navigation.

- **Style**
  - Updated pagination controls and navigation layout for a streamlined look.
  - Improved overall spacing, alignment, and form layouts for a cleaner, more user-friendly interface.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

yassirog committed:
feat: Expand Semester 4 sprints - Add logic and styles for semester 4 sprint handling, #62 ✨

Fixed #62

- Added logic to identify Semester 4 sprints based on sprint number.
- Applied custom styles for Semester 4 sprints, including specific height and visual adjustments.
- Removed unnecessary JS and moved it to the script

yassirog committed:
style: aligned show full agenda text next to button, fixes #73 💄

Fixed #73

- Deleted flex-direction column, so everything comes next to each other
- Align-items center, so the text would stay in the middle instead of above
- Added gap, for more whitespace between button and text

yassirog committed:
refactor: Clean code - Refactored, fixes #76 ♻️

Fixes #76,

Refactored the whole code:

- White-space
- No space behind css, example: display:flex;
- Commented (unused) code deleted
- JS code refactored

yassirog committed:
start with SVG components

components of SVG's

yassirog committed:
Revert "refactor: Clean code - Refactored, fixes #76 ♻️"

This reverts commit 705404315337343a38e04a9c24d4c8febc691d13.

yassirog committed:
refactor: Clean code - Refactored, fixes #76 ♻️

Fixes #76,

Refactored the whole code:

- White-space
- No space behind css, example: display:flex;
- Commented (unused) code deleted
- JS code refactored

yassirog committed:
feat: made components for SVG, fixes #77 ✨

fixed #77

- Made components for each SVG --> cleaner code

yassirog committed:
feat: svg component class prop, fixes #78 ✨

fixes #78

Made a Prop with a class in the SVG of the component,  because of this there weren't 2 SVG's needed.

Both were Arrows, 1 to the left and 1 to the right. I made a prop which gives 1 the class right-arrow with transformX -1. This makes the left arrow turn to the right.

yassirog committed:
feat: index.js used for cleaner import components, fixes #75 ✨

- Made a index.js file and imported all components to use it cleaner in other components.

yassirog committed:
bug: fix for not working show full agenda button, fixes #80 🐛

fixes #80

It was a small fix, it had to do with the last issue, i had duplicate imports in the index.js file which led to the button bugging.

joostf commented on src/lib/molecules/SprintLink.svelte:
```diff
@@ -80,6 +85,10 @@
     }
   }
 
+  li.semester4 {
```
Use a calc function to make the height more dynamic by multiply the default height by 3 (and subtracting redundant gap spacing)

joostf commented on src/lib/molecules/SprintLink.svelte:
```diff
@@ -1,20 +1,25 @@
+  let sprintClasses = [sprint.type]
+  if (active) sprintClasses.push('active')
+  if (past) sprintClasses.push('past')
+  if (semester4) sprintClasses.push('semester4')
```
This JavaScript is redundant.
Solve this in CSS by selecting the items in the last semester with a `project` class.

joostf commented on src/lib/atoms/Calendar.svelte:
```diff
@@ -1,16 +1,26 @@
 <script>
-  let { date } = $props()
+  let { date } = $props();
```
Don't close lines with a semicolon, as was added to the [FDND JavaScript conventions](https://github.com/fdnd-agency/.github/wiki/JavaScript-conventions) just now 😁

joostf commented on src/lib/molecules/Pagination.svelte:
```diff
@@ -1,70 +1,106 @@
     }
 
-    @media (min-width:47rem) {
+    @media (min-width: 47rem) {
```
Use container queries for component breakpoints

joostf commented on src/lib/molecules/StudentWork.svelte:
```diff
@@ -1,79 +1,74 @@
-          </li>
+  <section class="showcase {stargazer ? 'stargazer' : ''}">
+    <Heading title="Studentenwerk" />
+    <div class="gradient-container">
```
Better naming? What is a _gradient-container_?

joostf commented:
@yassirog I added some small non-breaking comments. Please derive (sub)issue from them and fix them in the next PR's

yassirog committed:
BUG: Deleted a criteria, which didn't let user open a sprint

- Deleted criteria html in queries/sprint.js, because this line of code led to not finding a sprint and making it NULL

yassirog committed:
Merge branch 'programma.release-candidate' into Yassir-design-tweaks
